### PR TITLE
Documentation for `zenml <stack_component> deploy` subcommand

### PR DESCRIPTION
**Pull request by strickvl**
Added:

- CLI docs
- a small documentation teaser for each relevant component flavor
- a central documentation section for information about this feature / functionality
- a note in the starter guide (as agreed per discord conversation) that this functionality is available

(Note that this PR is to be merged into the main implementation PR for this subcommand.)

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: docs/book/component-gallery/artifact-stores/gcp.md [hunk 1]
```diff
@@ -64,6 +64,22 @@ Depending on your use-case, however, you may also need to provide additional
 configuration parameters pertaining to [authentication](#authentication-methods)
 to match your deployment scenario.
 
+### Deploying with MLOps Stack Recipes
```
Comment (fa9r) [inline docs/book/component-gallery/artifact-stores/gcp.md:67]: Also no need to mention stack recipes here IMO

```suggestion
### Infrastructure Deployment
```

**Review by fa9r — CHANGES_REQUESTED** [review]
Looks good overall, the main page is very detailed, I love that you listed so much info here on which components exactly can be deployed and how they can be configured.

The main concern I have is that some of the links are split over multiple lines. Please check manually whether this works as expected or not.

I also think the intros are too wordy. IMO users don't need to know that this uses the stack recipes repo under the hood.

File: docs/book/advanced-guide/practical/stack-recipes.md [hunk 2]
```diff
@@ -81,7 +81,105 @@ In addition to the underlying `terraform` functionality, these commands also off
...
+
+You can also pass a region to deploy your resources to in the case of AWS and
+GCP recipes. For example, to deploy an S3 artifact store in the `us-west-2`
+region, you can run:
+
+```bash
+zenml artifact-store deploy s3_artifact_store --flavor=s3 --region=us-west-2
+```
+
+The default region is `eu-west-1` for AWS and `us-central1` for GCP.
```
[the remark this answers is not in the document]
Comment (strickvl) [reply docs/book/advanced-guide/practical/stack-recipes.md:164]: Haha yeah I wondered a bit about this. Maybe @wjayesh (who implemented it) can set them both to the same part of the world at least?

File: docs/book/advanced-guide/practical/stack-recipes.md [hunk 3]
```diff
@@ -81,7 +81,105 @@ In addition to the underlying `terraform` functionality, these commands also off
...
 - checks to ensure that you have all the binaries/tools installed for running a recipe.
 - extensive logs and error messages that guide you in case any of the recipes fails or misbehaves.
-- option to automatically import a ZenML stack out of the components created after deploying a stack recipe.
+- option to automatically import a ZenML stack out of the components created
+  after deploying a stack recipe.
+
+To learn more about what you can do with the ZenML CLI, please refer to the [CLI
+docs](https://apidocs.zenml.io/latest/cli/).
+
+### Deploying stack components directly
```
Comment (wjayesh) [inline docs/book/advanced-guide/practical/stack-recipes.md:90]:
```suggestion
## Deploying stack components directly
```
I would make this a higher heading and have it precede the "Integration with ZenML CLI" block to make it more visible. the purpose of that block is mostly to provide an understanding of what additional checks and capabilities are added by using the CLI on top of normal terraform.

File: docs/book/advanced-guide/practical/stack-recipes.md [hunk 4]
```diff
@@ -81,7 +81,105 @@ In addition to the underlying `terraform` functionality, these commands also off
...
+|   | tekton |
+|   | sagemaker |
+| Step Operator | sagemaker |
+| Secrets Manager | aws |
+|   | gcp |
+
+### Customizing the deployment of stack components
+
+You can pass configuration specific to the stack components as key-value
+arguments. If you don't provide a name, a random one is generated for you. For
```
Comment (wjayesh) [inline docs/book/advanced-guide/practical/stack-recipes.md:129]: headings for each component type to improve readability.
```suggestion
arguments. If you don't provide a name, a random one is generated for you. 

### Experiment Trackers
For
```

File: docs/book/advanced-guide/practical/stack-recipes.md [hunk 5]
```diff
@@ -81,7 +81,105 @@ In addition to the underlying `terraform` functionality, these commands also off
...
+zenml container-registry deploy aws_registry --flavor=aws --repo_name=my_repo
+```
+
+This is only useful for the AWS case since AWS requires a repository to be
+created before pushing images to it and the deploy command ensures that a
+repository with the name you provide is created. In case of GCP and other
+providers, you can choose the repository name at the same time as you are
+pushing the image via code. This is achieved through setting the `target_repo`
+attribute of the [the `DockerSettings` object](../pipelines/settings.md).
+
```
Comment (wjayesh) [inline docs/book/advanced-guide/practical/stack-recipes.md:155]: ### Other configuration

File: docs/book/advanced-guide/practical/stack-recipes.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (wjayesh) [reply docs/book/advanced-guide/practical/stack-recipes.md:164]: haha yes, this is from whatever defaults we set in the recipe for no reason whatsoever

File: docs/book/advanced-guide/practical/stack-recipes.md [hunk 6]
```diff
@@ -81,7 +81,105 @@ In addition to the underlying `terraform` functionality, these commands also off
...
+GCP recipes. For example, to deploy an S3 artifact store in the `us-west-2`
+region, you can run:
+
+```bash
+zenml artifact-store deploy s3_artifact_store --flavor=s3 --region=us-west-2
+```
+
+The default region is `eu-west-1` for AWS and `us-central1` for GCP.
+
+In the case of GCP components, it is *required* that you pass a project ID to
```
Comment (wjayesh) [inline docs/book/advanced-guide/practical/stack-recipes.md:166]: info or maybe warning is better
```suggestion
{% hint style="info" %} 
In the case of GCP components, it is *required* that you pass a project ID to
```

File: docs/book/component-gallery/artifact-stores/gcp.md [hunk 7]
```diff
@@ -64,6 +64,22 @@ Depending on your use-case, however, you may also need to provide additional
...
+
+A GCS Artifact Store can be deployed using our MLOps Stack Recipe
+directly from the CLI and without the need to go through a separate workflow
+involving stack recipes:
```
[the remark this answers is not in the document]
Comment (strickvl) [reply docs/book/component-gallery/artifact-stores/gcp.md:71]: Fixed everywhere + yes good point.

**Review by wjayesh — CHANGES_REQUESTED** [review]

File: docs/book/advanced-guide/practical/stack-recipes.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (wjayesh) [reply docs/book/advanced-guide/practical/stack-recipes.md:164]: okay, I'll set them both to europe-west

File: docs/book/advanced-guide/practical/stack-recipes.md (hunk 3 above)
Comment (strickvl) [reply docs/book/advanced-guide/practical/stack-recipes.md:90]: Got it. Moved now.

File: docs/book/advanced-guide/practical/stack-recipes.md (hunk 6 above)
Comment (strickvl) [reply docs/book/advanced-guide/practical/stack-recipes.md:160]: Done.

**Review by wjayesh — APPROVED** [review]
LGTM! just one final change in line 17. can you remove the hint? we don't need the `zenml[stacks]` requirement anymore.

**Comment by strickvl** [comment]
@wjayesh good catch. removed it.

File: docs/book/component-gallery/artifact-stores/gcp.md [hunk 8]
```diff
@@ -64,6 +64,22 @@ Depending on your use-case, however, you may also need to provide additional
...
+
+```shell
+zenml artifact-store deploy gcs_artifact_store --flavor=gcp ...
+```
+
+You can pass other configuration specific to the stack components as key-value
+arguments. If you don't provide a name, a random one is generated for you. For
+more information about how to work use the CLI for this, please refer to [the
+dedicated documentation
+section](../../advanced-guide/practical/stack-recipes.md#deploying-stack-components-directly).
```
[the remark this answers is not in the document]
Comment (fa9r) [reply docs/book/component-gallery/artifact-stores/gcp.md:79]: Ok fine by me then :D

**Review by fa9r — APPROVED** [review]
📝📚📖🗂️💡
